runtime/major_gc.c

```diff
@@ -1331,6 +1355,92 @@ static intnat ephe_sweep (caml_domain_state* domain_state, intnat budget)
   return budget;
 }
 
+static void request_mark_phase (void)
+{
+  if (caml_gc_phase == Phase_sweep_main &&
+      atomic_load_relaxed(&caml_gc_mark_phase_requested) == 0)
+    atomic_store_release(&caml_gc_mark_phase_requested, 1);
+}
+
+void caml_mark_roots_stw (int participant_count,
+                          caml_domain_state** barrier_participants)
+{
+  if (caml_gc_phase != Phase_sweep_main)
+    return;
+
+  enum global_roots_status {
+    WORK_UNSTARTED,
```

Comment:
What does `UNSTARTED` mean? `PAUSED`?

Reply:
It means that the scan of the global roots has not yet started. (This is unchanged from trunk: it used to happen in `cycle_major_heap_from_stw_single` / `stw_cycle_all_domains`, but has moved)

Reply:
UNSTARTED means that no domain has yet started marking the global roots. I've now renamed everything around here and added comments.